### PR TITLE
ENH: Add optional argument index to pd.melt to maintain index values

### LINKED ISSUE #17440: Index gets lost when DataFrame melt method is used
#### Index gets lost when DataFrame melt method is used

```python
import pandas as pd
import numpy as np
df = pd.DataFrame({"Numbers_1":range(0,3),
                   "Numbers_2":range(3,6),
                   "Letters":["A","B","C"]})
df.set_index("Letters",inplace=True)
print(df)
```
         
Letters | Numbers_1 |  Numbers_2             
| ------| ------| ------| 
A        |        0      |    3
B        |        1      |    4
C        |        2      |    5
```python

df_melted = df.melt()
print(df_melted)
```
 .     |variable | value
| ------| ------| ------| 
0 | Numbers_1   |   0
1 | Numbers_1   |   1
2 | Numbers_1   |   2
3 | Numbers_2   |   3
4 | Numbers_2   |   4
5 | Numbers_2   |   5

#### Problem description
When melting a dataframe, I expected the original index to be reused. However, the original index is lost in the melt method. This is probably meant by wesm's comment (# TODO: what about the existing index?)
https://github.com/pandas-dev/pandas/blob/133a2087d038da035a57ab90aad557a328b3d60b/pandas/core/reshape/reshape.py#L715 

#### Expected Output
I would expect something like
```python
n_row,n_col = df.shape
index_melted = list(df.index.get_values())*n_col
melt_id = list(np.arange(n_col).repeat(n_row))
temp = list(zip(*[index_melted,melt_id]))

index_melted_uniq = pd.MultiIndex.from_tuples(temp,names=[df.index.names[0], 'melt_id'])
index_numbers = list(range(df.shape[1]))*n_row

data = {'variable':df.columns.repeat(n_row),
        "value":df.values.ravel('F')}

df_expected = pd.DataFrame(data,columns = ["variable","value"], index=index_melted_uniq)
print(df_expected)
```     
Letters | melt_id | variable | value
|------|------|------|------|
A    |   0    |    Numbers_1   |   0
B    |   0     |   Numbers_1   |   1
C   |    0    |    Numbers_1   |   2
A   |    1    |    Numbers_2   |   3
B   |    1    |    Numbers_2   |   4
C   |    1   |     Numbers_2   |   5

Where Letters and melt_id are two multiindex levels and variable and value are actual columns.

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: d0f62c2816ada96a991f5a624a52c9a4f09617f7
python: 3.6.2.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.21.0.dev+420.gd0f62c2
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.2.2.post20170724
Cython: 0.26
numpy: 1.13.1
scipy: None
pyarrow: None
xarray: None
IPython: 6.1.0
sphinx: 1.6.3
patsy: None
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by Rik-de-Kort**
Finishing up a stale PR idea: #28859 and #17459

 Has some tests and better code.
I think it's fair to duplicate the index values and not bend over backwards to maintain uniqueness like in previous iterations.

Apologies for the mess, it was a quick job and I didn't want to spend an hour fiddling with the commits.

Finally, I deleted some ignore type comments for mypy because the commits weren't going on my system. Is there some other fix for that? Other than that I think it's good to go.

- [X] closes #17440
- [X] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry
- [x] Reconsider API design
- [x] Add usage examples in whatsnew, docstring, and reshaping.rst.

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @Rik-de-Kort for the PR.

> Finishing up a stale PR idea, here: #28859

The original work is #17459

Can you review that PR and also address any comments raised there that are relevant to this PR. e.g.

@TomAugspurger wrote https://github.com/pandas-dev/pandas/pull/17459#issuecomment-327818758

> For the docs, it'll need a whatsnew note in doc/source/whatsnew/v0.21.0.txt, prose docs in doc/source/reshaping.rst. It'd also be nice to have an example in the docstring.

can you update reshaping.rst and add examples to docstring. The release note should probably also include examples demonstrating the new functionality.

**Comment by Rik-de-Kort** [comment]
Thanks for commenting, @simonjayhawkins I couldn't find that PR when I was looking for it!

* I have renamed the kwarg to `index` rather than `keep_index` as suggested by @jreback.
* I have added an example to the docstring and reshaping.rst.

I don't think the whatsnew needs more prose about the change. It's use is straightforward: as opposed to the previous PR's it does the simplest thing imaginable, namely keep the original index values around in their original form.  
Speaking from an end user perspective (I use pandas a lot in my work), I think this is what I would like the API to be, but of course I'm open to discussion.  
One alternative would be to create a unique index using the old index values and `id_vars` as in the previous PRs. However, that is rather unwieldy in terms of code (it's bad enough to have to handle the case for the multiindex as it is), and also not very consistent with the behaviour when `index=False` because that keeps `id_vars` around as a column rather than as part of the index.

File: doc/source/user_guide/reshaping.rst [hunk 1]
```diff
@@ -296,6 +296,18 @@ For instance,
    cheese.melt(id_vars=['first', 'last'])
    cheese.melt(id_vars=['first', 'last'], var_name='quantity')
 
+Optionally, original index values can be kept around using the ``index`` parameter.
+
```
Comment (jreback) [inline doc/source/user_guide/reshaping.rst:300]: add a version added tag here

you can make this a subsection as well

File: doc/source/user_guide/reshaping.rst [hunk 2]
```diff
@@ -296,6 +296,18 @@ For instance,
...
+.. ipython:: python
+
+   cheese = pd.DataFrame({'first': ['John', 'Mary'],
+                          'last': ['Doe', 'Bo'],
+                          'height': [5.5, 6.0],
+                          'weight': [130, 150]})
+   cheese
+   cheese.melt(id_vars=['first', 'last'], index=True)
```
Comment (jreback) [inline doc/source/user_guide/reshaping.rst:308]: put the actual operations in a separate i python block (they render better)

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -99,7 +99,7 @@ Other enhancements
   ``df.to_csv(path, compression={'method': 'gzip', 'compresslevel': 1}``
   (:issue:`33196`)
 - :meth:`~pandas.core.groupby.GroupBy.transform` has gained ``engine`` and ``engine_kwargs`` arguments that supports executing functions with ``Numba`` (:issue:`32854`)
--
+- :meth: `melt` has gained an ``index`` argument that prevents the method from dropping the index.
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:102]: add the issue number

the ref to melt has an extra space (after the :)

Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:102]: say what the default is

File: pandas/core/frame.py [hunk 4]
```diff
@@ -6658,6 +6658,9 @@ def unstack(self, level=-1, fill_value=None):
         Name to use for the 'value' column.
     col_level : int or str, optional
         If columns are a MultiIndex then use this level to melt.
+    index : boolean, optional, default False
```
Comment (jreback) [inline pandas/core/frame.py:6661]: let’s rename to keep_index while longer is more descriptive

File: pandas/core/frame.py [hunk 5]
```diff
@@ -6708,6 +6711,17 @@ def unstack(self, level=-1, fill_value=None):
     1  b         B          3
     2  c         B          5
 
+    Original index values can be kept around:
+
+    >>> %(caller)sid_vars=['A'], value_vars=['B', 'C'], index=True)
```
Comment (jreback) [inline pandas/core/frame.py:6716]: not sure what u r doing here

File: pandas/core/reshape/melt.py [hunk 6]
```diff
@@ -115,7 +116,17 @@ def melt(
         # asanyarray will keep the columns as an Index
         mdata[col] = np.asanyarray(frame.columns._get_level_values(i)).repeat(N)
 
-    return frame._constructor(mdata, columns=mcolumns)
+    result = frame._constructor(mdata, columns=mcolumns)
+
+    if index:
+        new_index = np.tile(frame.index, K)
+
+        if isinstance(frame.index, MultiIndex):  # Index of tuples otherwise
```
Comment (jreback) [inline pandas/core/reshape/melt.py:124]: you might be able to use .from_arrays instead

File: pandas/tests/reshape/test_melt.py [hunk 7]
```diff
@@ -333,6 +333,29 @@ def test_melt_mixed_int_str_value_vars(self):
         expected = DataFrame({"variable": [0, "a"], "value": ["foo", "bar"]})
         tm.assert_frame_equal(result, expected)
 
+    def test_keep_index(self):
+        df = DataFrame({"foo": [0], "bar": [1]}, index=["first"])
```
Comment (jreback) [inline pandas/tests/reshape/test_melt.py:337]: add the issue number as. comment

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/reshaping.rst (hunk 1 above)
Comment (Rik-de-Kort) [reply doc/source/user_guide/reshaping.rst]: Done on both accounts.

File: doc/source/user_guide/reshaping.rst (hunk 2 above)
Comment (Rik-de-Kort) [reply doc/source/user_guide/reshaping.rst]: I have done this, but the rest of the documentation doesn't do it, and I'm not sure I like the look. Maybe I misunderstood?

File: doc/source/whatsnew/v1.1.0.rst (hunk 3 above)
Comment (Rik-de-Kort) [reply doc/source/whatsnew/v1.1.0.rst]: Done

File: pandas/core/frame.py (hunk 5 above)
Comment (Rik-de-Kort) [reply pandas/core/frame.py]: Adding an example to the docstring. Passes docstring verification.

File: pandas/core/reshape/melt.py (hunk 6 above)
Comment (Rik-de-Kort) [reply pandas/core/reshape/melt.py:124]: It seems the natural "arrangement" for MultiIndex values is in tuples form. There is the `codes`-property but that doesn't work because it's not the original index values. I think using from_tuples is preferable in this case.

File: pandas/core/reshape/melt.py [hunk 8]
```diff
@@ -115,7 +116,17 @@ def melt(
         # asanyarray will keep the columns as an Index
         mdata[col] = np.asanyarray(frame.columns._get_level_values(i)).repeat(N)
 
-    return frame._constructor(mdata, columns=mcolumns)
+    result = frame._constructor(mdata, columns=mcolumns)
+
+    if keep_index:
+        new_index = np.tile(frame.index, K)
```
Comment (simonjayhawkins) [inline pandas/core/reshape/melt.py:122]: Index types (e.g. Categorical) and index names are not retained?

Comment (Rik-de-Kort) [reply pandas/core/reshape/melt.py:122]: Good observation, added.

File: doc/source/user_guide/reshaping.rst [hunk 9]
```diff
@@ -312,6 +313,27 @@ user-friendly.
   dft
   pd.wide_to_long(dft, ["A", "B"], i="id", j="year")
 
+
+Keeping the index around
+~~~~~~~~~~~~~~~~~~~~~~~~
+When transforming a DataFrame using :func:`~pandas.melt`, original index values can be kept around by setting the ``keep_index`` parameter to True. This will however duplicate the index values.
```
Comment (TomAugspurger) [inline doc/source/user_guide/reshaping.rst:319]:
```suggestion

When transforming a DataFrame using :func:`~pandas.melt`, original the index values can be kept around by setting the ``keep_index`` parameter to True. This will however duplicate the index values.
```

File: pandas/core/frame.py [hunk 10]
```diff
@@ -6811,6 +6811,9 @@ def unstack(self, level=-1, fill_value=None):
         Name to use for the 'value' column.
     col_level : int or str, optional
         If columns are a MultiIndex then use this level to melt.
+    keep_index : bool, optional, default False
+            If True, the original index is reused.
+            There is no uniqueness guarantee.
```
Comment (TomAugspurger) [inline pandas/core/frame.py:6816]:
```suggestion
        If True, the original index is retained. Index labels will be repeated
        as necessary.
```

**Review by TomAugspurger — APPROVED** [review]
Looks nice.

The repeated values from `keep_index=True` naturally makes me think that we should form a MultIndex out of the `id_vars` and the original index, but that can be a separate issue.

**Comment by Rik-de-Kort** [comment]
Some issues with the CI mysteriously failing. Seems resolved now.

@jreback @simonjayhawkins if you have time, could you please review? Of course understandable if priorities lie elsewhere :)

File: doc/source/user_guide/reshaping.rst [hunk 11]
```diff
@@ -312,6 +313,28 @@ user-friendly.
   dft
   pd.wide_to_long(dft, ["A", "B"], i="id", j="year")
 
+
+Keeping the index around
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+When transforming a DataFrame using :func:`~pandas.melt`, original the index values can be kept around by setting the ``keep_index`` parameter to True. This will however duplicate the index values.
```
Comment (jreback) [inline doc/source/user_guide/reshaping.rst:320]: add a versionadded 1.1

Comment (jreback) [reply doc/source/user_guide/reshaping.rst:320]: move from below

File: doc/source/user_guide/reshaping.rst [hunk 12]
```diff
@@ -312,6 +313,28 @@ user-friendly.
...
+
+.. ipython:: python
+
+   index = pd.MultiIndex.from_tuples([('person', 'first'), ('person', 'second')])
```
Comment (jreback) [inline doc/source/user_guide/reshaping.rst:324]: this should be after the melt example above.

File: doc/source/whatsnew/v1.1.0.rst [hunk 13]
```diff
@@ -146,11 +146,12 @@ Other enhancements
   This can be used to set a custom compression level, e.g.,
   ``df.to_csv(path, compression={'method': 'gzip', 'compresslevel': 1}``
   (:issue:`33196`)
+- :meth:`~pandas.core.groupby.GroupBy.transform` has gained ``engine`` and ``engine_kwargs`` arguments that supports executing functions with ``Numba`` (:issue:`32854`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst:149]: why is this changed? (the transform note)

File: pandas/core/frame.py [hunk 14]
```diff
@@ -6811,6 +6811,9 @@ def unstack(self, level=-1, fill_value=None):
         Name to use for the 'value' column.
     col_level : int or str, optional
         If columns are a MultiIndex then use this level to melt.
+    keep_index : bool, optional, default False
+        If True, the original index is retained. Index labels will be repeated
```
Comment (jreback) [inline pandas/core/frame.py:6815]: needs a versionadded 1.1

File: pandas/core/reshape/melt.py [hunk 15]
```diff
@@ -115,7 +116,19 @@ def melt(
...
+
+        if isinstance(frame.index, MultiIndex):
+            new_index = MultiIndex.from_tuples(new_index, names=frame.index.names)
+        else:
+            new_index = Index(new_index, dtype=frame.index.dtype, name=frame.index.name)
+
+        result = result.set_index(new_index)
```
Comment (jreback) [inline pandas/core/reshape/melt.py:129]: result.index = new_index is enough here (its already a copy)

File: pandas/tests/reshape/test_melt.py [hunk 16]
```diff
@@ -333,6 +333,42 @@ def test_melt_mixed_int_str_value_vars(self):
...
+        tm.assert_frame_equal(result, expected)
+
+    def test_keep_index_name_and_type(self):
+        # GH 17440
+        index = pd.Index(["foo", "bar"], dtype="category", name="baz")
+        df = DataFrame({"x": [0, 1], "y": [2, 3]}, index=index)
+        result = melt(df, keep_index=True)
+
+        expected_index = pd.Index(["foo", "bar"] * 2, dtype="category", name="baz")
+        tm.assert_index_equal(result.index, expected_index)
```
Comment (jreback) [inline pandas/tests/reshape/test_melt.py:370]: use assert_frame_equal and construct the expected

File: pandas/core/frame.py [hunk 17]
```diff
@@ -6811,6 +6811,9 @@ def unstack(self, level=-1, fill_value=None):
         Name to use for the 'value' column.
     col_level : int or str, optional
         If columns are a MultiIndex then use this level to melt.
+    keep_index : bool, optional, default False
```
Comment (jreback) [inline pandas/core/frame.py:6814]: this needs to be called ignore_index as we use this everywhere else, default True i think (as that is what we do now)

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/reshaping.rst [hunk 18]
```diff
@@ -312,6 +331,8 @@ user-friendly.
   dft
   pd.wide_to_long(dft, ["A", "B"], i="id", j="year")
 
+
+
```
Comment (simonjayhawkins) [inline doc/source/user_guide/reshaping.rst:335]: why are these blank lines added?

File: doc/source/whatsnew/v1.1.0.rst [hunk 19]
```diff
@@ -220,6 +220,8 @@ Other enhancements
   This can be used to set a custom compression level, e.g.,
   ``df.to_csv(path, compression={'method': 'gzip', 'compresslevel': 1}``
   (:issue:`33196`)
+- :meth:`~pandas.core.groupby.GroupBy.transform` and :meth:`~pandas.core.groupby.GroupBy.aggregate` has gained ``engine`` and ``engine_kwargs`` arguments that supports executing functions with ``Numba`` (:issue:`32854`, :issue:`33388`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.1.0.rst:223]: why is this added? from resolving a merge conflict maybe?

Comment (Rik-de-Kort) [reply doc/source/whatsnew/v1.1.0.rst]: Yeah that's why I put it there, but it seems to be unnecessary.

**Comment by Rik-de-Kort** [comment]
Does the CI always have this many issues?

File: pandas/core/frame.py [hunk 20]
```diff
@@ -6921,6 +6921,11 @@ def unstack(self, level=-1, fill_value=None):
         Name to use for the 'value' column.
     col_level : int or str, optional
         If columns are a MultiIndex then use this level to melt.
+    ignore_index : bool, optional, default True
+        If False, original index is ignored. If True, the original index is retained.
```
Comment (simonjayhawkins) [inline pandas/core/frame.py:6925]: is this correct? if ignore_index=False, the index is ignored?

Comment (Rik-de-Kort) [reply pandas/core/frame.py]: Oh my word... Thanks for being so thorough!

File: pandas/core/frame.py [hunk 21]
```diff
@@ -7008,7 +7008,6 @@ def unstack(self, level=-1, fill_value=None):
              b   4.0
         dtype: float64
 
-        >>> s.unstack(level=-1)
```
Comment (simonjayhawkins) [inline pandas/core/frame.py:7011]: @Rik-de-Kort I don't think you intended to remove this.

 the doc error is `/home/runner/work/pandas/pandas/pandas/core/frame.py:docstring of pandas.DataFrame.unstack:55: WARNING: Block quote ends without a blank line; unexpected unindent.`

so could be related.

Comment (Rik-de-Kort) [reply pandas/core/frame.py]: ahhh thanks! all checks passing now, finally!

File: doc/source/user_guide/reshaping.rst [hunk 22]
```diff
@@ -296,6 +296,25 @@ For instance,
    cheese.melt(id_vars=['first', 'last'])
    cheese.melt(id_vars=['first', 'last'], var_name='quantity')
 
+When transforming a DataFrame using :func:`~pandas.melt`, original the index values can be kept around by setting the ``ignore_index`` parameter to ``False``. This will however duplicate the index values.
```
Comment (simonjayhawkins) [inline doc/source/user_guide/reshaping.rst:299]:
```suggestion
When transforming a DataFrame using :func:`~pandas.melt`, the original index values can be kept around by setting the ``ignore_index`` parameter to ``False``. This will however duplicate the index values.
```

File: doc/source/user_guide/reshaping.rst [hunk 23]
```diff
@@ -296,6 +296,25 @@ For instance,
...
+
+   index = pd.MultiIndex.from_tuples([('person', 'first'), ('person', 'second')])
+   cheese = pd.DataFrame({'first': ['John', 'Mary'],
+                          'last': ['Doe', 'Bo'],
+                          'height': [5.5, 6.0],
+                          'weight': [130, 150]},
+                         index=index)
+   cheese
+
+.. ipython:: python
```
Comment (simonjayhawkins) [inline doc/source/user_guide/reshaping.rst:311]: no need to repeat the directive here, as it creates an extra box and splits the example

File: doc/source/user_guide/reshaping.rst [hunk 24]
```diff
@@ -296,6 +296,25 @@ For instance,
    cheese.melt(id_vars=['first', 'last'])
    cheese.melt(id_vars=['first', 'last'], var_name='quantity')
 
+When transforming a DataFrame using :func:`~pandas.melt`, original the index values can be kept around by setting the ``ignore_index`` parameter to ``False``. This will however duplicate the index values.
+
+.. ipython:: python
+
+   index = pd.MultiIndex.from_tuples([('person', 'first'), ('person', 'second')])
```
Comment (simonjayhawkins) [inline doc/source/user_guide/reshaping.rst:303]: this example is a bit confusing with first and second since first and last are columns and used as id_vars latter in the example. maybe just 1 and 2 or A and B?

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @Rik-de-Kort almost there. generally lgtm. just a couple more comments.

File: doc/source/user_guide/reshaping.rst [hunk 25]
```diff
@@ -296,6 +296,25 @@ For instance,
...
+                          'height': [5.5, 6.0],
+                          'weight': [130, 150]},
+                         index=index)
+   cheese
+
+.. ipython:: python
+
+   cheese.melt(id_vars=['first', 'last'], ignore_index=False)
+
+.. versionadded:: 1.1.0
```
Comment (jreback) [inline doc/source/user_guide/reshaping.rst:315]: can you put this at the beginning of the section

File: pandas/core/frame.py [hunk 26]
```diff
@@ -7086,6 +7086,7 @@ def melt(
         var_name=None,
         value_name="value",
         col_level=None,
+        ignore_index=True,
```
Comment (jreback) [inline pandas/core/frame.py:7089]: can you type ( i know nothing else is, but at least this is a start)

File: pandas/core/reshape/melt.py [hunk 27]
```diff
@@ -121,7 +121,19 @@ def melt(
         # asanyarray will keep the columns as an Index
         mdata[col] = np.asanyarray(frame.columns._get_level_values(i)).repeat(N)
 
-    return frame._constructor(mdata, columns=mcolumns)
+    result = frame._constructor(mdata, columns=mcolumns)
+
+    if not ignore_index:
+        new_index = np.tile(frame.index, K)
```
Comment (jreback) [inline pandas/core/reshape/melt.py:127]: both MI and Index already have a .repeat() method, I think we could add a .tile() method to make this easier. (or just use repeat)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 26 above)
Comment (Rik-de-Kort) [reply pandas/core/frame.py:7089]: Of course!

File: pandas/core/reshape/melt.py (hunk 27 above)
Comment (Rik-de-Kort) [reply pandas/core/reshape/melt.py:127]: `Index(["foo", "bar"]).repeat(2)` yields `Index(['foo', 'foo', 'bar', 'bar'], dtype='object')`, where as `np.tile(["foo", "bar"])` yields `array(['foo', 'bar', 'foo', 'bar', 'foo', 'bar'], dtype=object)`. The latter corresponds to the layout used in `melt` so it's very not trivial to use repeat instead of tile.

I tried having a look at implementing `tile` on indices but then I would also have to do it for multiindices and document it and tests, and argument validation which I've never even looked at before and I think it's a big hassle that I will not undertake.

File: doc/source/user_guide/reshaping.rst [hunk 28]
```diff
@@ -296,6 +296,21 @@ For instance,
...
+.. ipython:: python
+
+   index = pd.MultiIndex.from_tuples([('person', 'A'), ('person', 'B')])
+   cheese = pd.DataFrame({'first': ['John', 'Mary'],
+                          'last': ['Doe', 'Bo'],
+                          'height': [5.5, 6.0],
+                          'weight': [130, 150]},
+                         index=index)
+   cheese
+   cheese.melt(id_vars=['first', 'last'], ignore_index=False)
```
Comment (jreback) [inline doc/source/user_guide/reshaping.rst:312]: can you add an example w/o passing ignore_index and indicate that the default is true (unless we already have that)?

e.g. put *another* ipython block starting L312 and then put the False case after

File: pandas/core/reshape/melt.py (hunk 27 above)
Comment (jreback) [reply pandas/core/reshape/melt.py:127]: ok fair enough

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. a doc comment, pls ping on green.

File: doc/source/user_guide/reshaping.rst (hunk 28 above)
Comment (Rik-de-Kort) [reply doc/source/user_guide/reshaping.rst:313]: @jreback it is green therefore ping

File: pandas/core/reshape/melt.py (hunk 27 above)
Comment (simonjayhawkins) [reply pandas/core/reshape/melt.py:127]: we already have _tile_compat in pandas\core\reshape\util.py. This may allow futher simplification here.

**Comment by Rik-de-Kort** [comment]
@jreback I though I gave you a ping, but I don't see it, so here it is!

File: pandas/core/reshape/melt.py (hunk 27 above)
Comment (TomAugspurger) [reply pandas/core/reshape/melt.py:127]: This converts to object dtype. Can you use `pandas.core.reshape.util._tile_compat`?:

Comment (TomAugspurger) [reply pandas/core/reshape/melt.py:127]: You should be able to remove the next section as well, since you won't be converting to an ndarray.

Comment (Rik-de-Kort) [reply pandas/core/reshape/melt.py]: Thanks, that simplifies the code a lot.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Rik-de-Kort

**Comment by Rik-de-Kort** [comment]
You're welcome
